lib/vnet/daemon/protocol_darwin.h

```diff
@@ -1,6 +1,17 @@
 #ifndef TELEPORT_LIB_VNET_DAEMON_PROTOCOL_DARWIN_H_
 #define TELEPORT_LIB_VNET_DAEMON_PROTOCOL_DARWIN_H_
 
+#import <Foundation/Foundation.h>
+
+static NSString *const VNEErrorDomain = @"com.Gravitational.Vnet.ErrorDomain";
```

Comment:
I saw the related articles, but I'm keeping this and the next comment (on VNEErrorCode), as I think they still have some merit.

We can do the appropriate conversions on the stop, in the ObjC code. LMKWYT.

Reply:
Oh yeah, that seems like a good idea to avoid copying the error domain between different languages. I think Obj-C will be just fine with passing something like `@(VNEErrorDomain)` as the error domain to `NSError`.

Reply:
I would add the literal constant to the header and call it done.

Reply:
That's what I wanted to do, but I'm having problems with this. Any of these two options:

```
const char *VNEErrorDomain = "com.Gravitational.Vnet.ErrorDomain";
```

```
const char * const VNEErrorDomain = "com.Gravitational.Vnet.ErrorDomain";
```

results in: 

```
/Users/rav/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.5.darwin-arm64/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
duplicate symbol '_VNEErrorDomain' in:
    /var/folders/kz/2d_vhlks21l3g0s4g5yytlwr0000gn/T/go-link-4167872705/000008.o
    /var/folders/kz/2d_vhlks21l3g0s4g5yytlwr0000gn/T/go-link-4167872705/000010.o
duplicate symbol '_VNEErrorDomain' in:
    /var/folders/kz/2d_vhlks21l3g0s4g5yytlwr0000gn/T/go-link-4167872705/000008.o
    /var/folders/kz/2d_vhlks21l3g0s4g5yytlwr0000gn/T/go-link-4167872705/000012.o
ld: 2 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

With this:

```
static const char* const VNEErrorDomain = "com.Gravitational.Vnet.ErrorDomain";
```

I get:

```
/Users/rav/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.5.darwin-arm64/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
Undefined symbols for architecture arm64:
  "_VNEErrorDomain", referenced from:
      _github.com/gravitational/teleport/lib/vnet/daemon._Cvar_VNEErrorDomain in go.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Finally, defining `extern` in the header file and then a separate `const` in an .m file seems to work.

Reply:
`extern` seems to be right call. From what I've tried a macro also works.